test/protocol_test.py

```diff
@@ -1274,18 +1274,30 @@ def test_one_source(self, dummy_protocol):
         p = dummy_protocol
         echo = p.ref("echo", None, "384-echo", discard=True)
         dest = p.ref("dest", None, "384-flat", discard=True)
+
+        # dead volume for source wells(15 uL) results in 0 available volume, not enough to transfer
+        with pytest.raises(RuntimeError):
+            p.acoustic_transfer(
+                echo.wells(0, 1).set_volume("2:microliter"),
+                dest.wells(0, 1, 2, 3),
+                "1:microliter",
+                one_source=True,
+            )
+
         p.acoustic_transfer(
-            echo.wells(0, 1).set_volume("2:microliter"),
-            dest.wells(0, 1, 2, 3),
-            "1:microliter",
+            echo.wells(0, 1, 2, 3).set_volume("60:microliter"),
+            dest.wells(list(range(0, 60))),
+            "2:microliter",
             one_source=True,
         )
+
         assert p.instructions[-1].data["groups"][0]["transfer"][-1][
             "from"
-        ] == echo.well(1)
+        ] == echo.well(2)
         assert p.instructions[-1].data["groups"][0]["transfer"][0]["from"] == echo.well(
             0
         )
```

Comment:
Do we need to know where the transfers came from or just that the volume was transferred and there was enough in the well? I think we may be able to get rid of these assertions.

Reply:
i think originally it was there to verify that the source wells were being changed if there was supposed to be a transfer from an empty well. in this case i guess its still doing something similar in that it is verifying that the second well will be used even though there is still volume remaining in the first source well

Reply:
but i think this could probably be tested more gracefully. the assertions are a bit clunky